packages/instant/src/components/buy_button.tsx

```diff
@@ -17,6 +19,7 @@ export interface BuyButtonProps {
     onBuyProcessing: (buyQuote: BuyQuote, txHash: string) => void;
     onBuySuccess: (buyQuote: BuyQuote, txHash: string) => void;
     onBuyFailure: (buyQuote: BuyQuote, txHash: string) => void;
+    validateWalletBeforeBuy: (buyQuote: BuyQuote, takerAddress: string | undefined) => Promise<boolean>;
```

Comment:
we usually just write `takerAddress?: string` for optional params we want to add an "async" suffix to the prop name to signal that it is awaitable, i.e. `validateAddressBeforeBuyAsync`